pyfar/dsp/filter/band_filter.py

```diff
@@ -450,3 +450,68 @@ def crossover(signal, N, frequency, sampling_rate=None):
         # return the filtered signal
         signal_filt = filt.process(signal)
         return signal_filt
+
+
+def notch(signal, center_frequency, quality, sampling_rate=None):
+    """
+    Create and apply a second order IIR notch filter.
+
+    A notch filter has a magnitude of 0 (-infinity dB) at the center frequency
+    and a magnitude of 1 (0 dB) at almost all other frequencies. This is a
+    wrapper for ``scipy.signal.iirnotch``.
+
+    Parameters
+    ----------
+    signal : Signal, None
+        The Signal to be filtered. Pass ``None`` to create the filter without
+        applying it.
+    center_ frequency : number
+        Frequency in Hz at which the magnitude response will be 0
+        (-infinity dB).
+    quality : number
+        The quality is defined at the -3 dB points around the
+        `center_frequency` and is the inverse band width in octaves relative
+        to the `center_frequency`
+    sampling_rate : None, number
+        The sampling rate in Hz. Only required if `signal` is ``None``. The
+        default is ``None``.
+
+    Returns
+    -------
+    signal : Signal
```

Comment:
```suggestion
    result: Signal, FilterSOS
```
alright, same style in other filter types. maybe it is a bit confusing, if 2 outputs are listes, but just one of them is returned, maybe we should name it a bit more general and then describe what type is returned in which case. this doc looks to me like it always returns 2 things. We can discuss in the weekly

Reply:
True - but that would be a separate pull

Reply:
thx

Reply:
Yes, true. The correct way of handling this would be something like this

```suggestion
    output : Signal, FilterIIR
        The function returns a filtered version of the input signal if ``sampling_rate = None`` or the filter itself if ``signal = None``.
```

Analogous to 
```
signal : Signal, None
        The Signal to be filtered. Pass ``None`` to create the filter without  applying it.
```
a couple of lines above.

Reply:
I don't agree that this is a separate pull to be honest.